consensus/src/block_storage/block_store_and_lec_recovery_test.rs

```diff
@@ -0,0 +1,256 @@
+    config.execution.service =
+        ExecutionCorrectnessService::SpawnedProcess(RemoteExecutionService { server_address });
+
+    // Store the config
+    let mut config_path = TempPath::new();
+    config_path.persist();
+    config_path.create_as_file().unwrap();
+    config.save_config(&config_path).unwrap();
+
+    let execution_correctness_manager = ExecutionCorrectnessManager::new(&mut config);
+
+    let (coordinator_sender, _coordinator_receiver) = mpsc::unbounded();
+
+    let state_computer = Arc::new(ExecutionProxy::new(
+        execution_correctness_manager.client(),
+        Arc::new(StateSyncClient::new(coordinator_sender)),
+    ));
+
+    let (initial_data, qc) = get_initial_data_and_qc(&*db);
+
+    let mut inserter = TreeInserter::new_with_store(
+        ValidatorSigner::new(
+            config.validator_network.unwrap().peer_id(),
+            config
+                .test
+                .as_mut()
+                .unwrap()
+                .consensus_keypair
+                .as_mut()
+                .unwrap()
+                .take_private()
+                .unwrap(),
+        ),
+        Arc::new(BlockStore::new(
+            Arc::new(EmptyStorage::new()),
+            initial_data,
+            state_computer,
+            10, // max pruned blocks in mem
+            Arc::new(SimulatedTimeService::new()),
+        )),
+    );
+    let block_store = inserter.block_store();
+    let genesis = block_store.root();
+    let genesis_block_id = genesis.id();
+    let genesis_block = block_store
+        .get_block(genesis_block_id)
+        .expect("genesis block must exist");
+
+    //       ╭--> A1--> A2
+    // Genesis--> B1
+    let a1 = inserter.insert_block_with_qc(qc.clone(), &genesis_block, 1);
+    let a2 = inserter.insert_block(&a1, 2, None);
+    let b1 = inserter.insert_block_with_qc(qc, &genesis_block, 4);
+
+    // Crash LEC.
+    drop(execution_correctness_manager);
+
+    config = NodeConfig::load(&config_path).unwrap();
+    // Restart LEC and make sure we can continue to append to the current tree.
+    let _execution_correctness_manager = ExecutionCorrectnessManager::new(&mut config);
+
+    //       ╭--> A1--> A2--> A3
+    // Genesis--> B1--> B2
+    //             ╰--> C1
+    let _a3 = inserter.insert_block(&a2, 3, Some(genesis.block_info()));
+    let _b2 = inserter.insert_block(&b1, 5, None);
+    let _c1 = inserter.insert_block(&b1, 6, None);
+}
+
+#[test]
+fn test_block_store_restart() {
+    // Start storage service
+    let (mut config, _handle, db) = start_storage_service();
+
+    let server_port = utils::get_available_port();
+    let server_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), server_port);
+
+    // Start LEC service as a process.
+    config.execution.service =
+        ExecutionCorrectnessService::SpawnedProcess(RemoteExecutionService { server_address });
+
+    // Store the config
+    let mut config_path = TempPath::new();
+    config_path.persist();
```

Comment:
why persist?

Reply:
Otherwise it give me some random serde error if I use the same config.

Reply:
sounds weird